pkg/controller/servicemesh/webhookca/controller.go

```diff
@@ -97,7 +97,7 @@ var autoRegistrationMap = map[string]CABundleSource{
 			},
 			{
 				SecretName: istiodCertManagerSecretName,
-				Key:        common.IstiodTLSSecretCertKey,
+				Key:        common.IstiodTLSSecretCertCAKey,
```

Comment:
Should this be done also for mutating web hook?

Reply:
It can be although it does not seem like the OSSM2 operator tries to insert the istiod tls cert into the OSSM3 webhook. I'm guessing because the ossm2 webhook has a different naming scheme:
```
istio-sidecar-injector-ossm3     2          20h
istiod-basic-istio-system        1          22h
```

We should probably update for `cacerts` as well since that will likely have the same problem.

Reply:
@FilipB I updated the rest of the istiod webhooks and updated the tests as well